engine/transaction/outlines/internal/inputs/testabilities/assertions_inputs_selector.go

```diff
@@ -0,0 +1,78 @@
+package testabilities
+
+import (
+	"testing"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/database"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type InputsSelectorAssertions interface {
+	WithoutError(error) SuccessfullySelectedInputsAssertions
+}
+
+type SuccessfullySelectedInputsAssertions interface {
+	SelectedInputs(inputs []*database.UserUtxos) SelectedInputsAssertions
+}
+
+type SelectedInputsAssertions interface {
+	AreEmpty()
+	ComparingTo(inputs []*database.UserUtxos) ComparingSelectedInputsAssertions
+}
+
+type ComparingSelectedInputsAssertions interface {
+	AreEntries(expectedIndexes []int)
+}
+
+type assertion struct {
+	t               testing.TB
+	require         *require.Assertions
+	assert          *assert.Assertions
+	actual          []*database.UserUtxos
+	comparingSource []*database.UserUtxos
+}
+
+func newAssertions(t testing.TB) InputsSelectorAssertions {
+	return assertion{
+		t:       t,
+		require: require.New(t),
+		assert:  assert.New(t),
+	}
+}
+
+func (a assertion) WithoutError(err error) SuccessfullySelectedInputsAssertions {
+	a.t.Helper()
+	a.require.NoError(err)
+	return a
+}
+
+func (a assertion) SelectedInputs(inputs []*database.UserUtxos) SelectedInputsAssertions {
+	a.t.Helper()
```

Comment:
So you decided to give `Helper` a chance ?
But now, we will have inconsistent test helpers (with and without `t.Helper` call)

Reply:
meh,
Actually I don't see any difference, so for me it's a minor thing to keep consistency in this matter.

Reply:
Basically, there is a difference.  Take a look at the following example:

```
import "testing"

func FuncMarkedAsTestHelper(t *testing.T) {
	t.Helper()
	t.Fatal("something bad happend") // line 7
}

func FuncNotMarkedAsTestHelper(t *testing.T) {
	t.Fatal("something bad happend") // line 11
}

func Test(t *testing.T) {
	FuncNotMarkedAsTestHelper(t) // line 15
}
```

**Test output running FuncNotMarkedAsTestHelper call:** 
--- FAIL: Test (0.00s)    .../dummy_test.go:**11**: something bad happend
 
**Test output running FuncNotMarkedAsTestHelper call:** 
--- FAIL: Test (0.00s)    .../dummy_test.go:**15**: something bad happend

You can see the line reference is different.  Marking functions as `t.Helper()` helps in navigation and debugging by ensuring that when a test fails (due to `t.Fatal`, `t.Fatalf`, or a `panic`), the failure report points to the line in the test where the helper function was called, rather than the line inside the helper function itself.

Reply:
1. I know what is the difference, but when using testify, you're getting the whole stack and the file name is a littlebit secondary
2. I'm not sure if we we should somehow distiguish the usage of the t.Helper between env arrangement (when the fixtures are preapred and everything is established and the assertions. Because if something fails durring setup I would rather want to search for the place where it really fail, not the code that call that place, while when making assertions I'm rather interested in what were the values that wasn't expected and at what point in the test exectution it happens.

if you want to push for t.Helpers, I would rather move this discussion out of this PR, and agree on the standard within whole team and adjust existing code in some PR.

Reply:
I would resolve this conversation 🙈